### ISSUE TITLE
when the two doc do not have common word vocab, do this model work?

### ISSUE THREAD

**Issue by guotong1988** [faneshion/MatchZoo#88]
word vocab: 1 2 3 4 5 6 7 8 9
doc A : 1 2 or 2 3 or 1 3
doc B : 4 5 6 or 7 8 9 or 4 6 8 or 5 7 9

@faneshion @yangliuy @pl8787 @zhouzhouyang520 @HouJP @uduse

**Comment by thiziri** [comment]
I think that since the different models make use of word embeddings (that enable capturing semantic relatedness) it doesn't matter if the compared two texts don't have any word in common. 
ps: Thanks for asking and answering questions in English, please, to let everyone understand discussions.

**Comment by bwanglzu** [comment]
Believe if the model was based on word embeddings, it will capture the semantic relationship. However, some models might not build based on embeddings, such as "Learning Deep Structured Semantic Models
for Web Search using Clickthrough Data", it's architecture is something like: `raw terms` -> `word hashing (tri-grams)` -> `2*300Dimentional Dense Layers ` -> `1*128Dimentional Dense Layers` -> `Cosine SImilarity` -> `Softmax with Gamma`.

**Comment by guotong1988** [comment]
我发现`DRMM_TKS`这个模型在有共现词的时候很准，没有共现词的时候根本不行啊，我这出什么问题了？
I find the drmm_tks model mainly work when there are common words, but do not work when there are not common words.

```

{
  "net_name": "DRMM_TKS",
  "global":{
      "model_type": "PY",
      "weights_file": "/home/gt/MatchZoo/nlpcc/save_model/drmm_tks_ranking.weights",
      "save_weights_iters": 100,
      "num_iters": 2500,
      "display_interval": 100,
      "test_weights_iters": 2500,
      "optimizer": "adam",
      "learning_rate": 0.0001
  },
  "inputs": {
    "share": {
        "text1_corpus": "/home/gt/MatchZoo/nlpcc/corpus_preprocessed.txt",
        "text2_corpus": "/home/gt/MatchZoo/nlpcc/corpus_preprocessed.txt",
        "use_dpool": false,
        "embed_size": 50,
        "vocab_size": 3004,
        "train_embed": true,
        "target_mode": "ranking",
        "text1_maxlen": 50,
        "text2_maxlen": 50
    },
    "train": {
        "input_type": "PairGenerator", 
        "phase": "TRAIN",
        "use_iter": false,
        "query_per_iter": 50,
        "batch_per_iter": 5,
        "batch_size": 100,
        "relation_file": "/home/gt/MatchZoo/nlpcc/relation_train.txt"
    },
    "valid": {
        "input_type": "ListGenerator", 
        "phase": "EVAL",
        "batch_list": 20,
        "relation_file": "/home/gt/MatchZoo/nlpcc/relation_train.txt"
    },
    "test": {
        "input_type": "ListGenerator", 
        "phase": "EVAL",
        "batch_list": 20,
        "relation_file": "/home/gt/MatchZoo/nlpcc/relation_test_real.txt"
    },
    "predict": {
        "input_type": "ListGenerator", 
        "phase": "PREDICT",
        "batch_list": 20,
        "relation_file": "/home/gt/MatchZoo/nlpcc/relation_test_real.txt"
    }
  },
  "outputs": {
    "predict": {
      "save_format": "TREC",
      "save_path": "drmm_tks_ranking_predict.txt"
    }
  },
  "model": {
    "model_path": "/home/gt/MatchZoo/nlpcc/save_model",
    "model_py": "drmm_tks.DRMM_TKS",
    "setting": {
        "topk": 50,
        "num_layers": 2,
        "hidden_sizes": [100, 1],
        "dropout_rate": 0.5
    }
  },
  "losses": [ 
    {
       "object_name": "rank_hinge_loss" ,
       "object_params": {
            "margin": 0.5
       }
    }
  ],
  "metrics": [ "ndcg@1", "ndcg@3", "map" ]
}


```

**Comment by guotong1988** [comment]
something useful:
https://github.com/faneshion/MatchZoo/issues/91
